scripts/lib/CIME/XML/env_batch.py

```diff
@@ -496,16 +499,23 @@ def _submit_single_job(self, case, job, dep_jobs=None, no_batch=False,
                "Unable to determine the correct command for batch submission.")
         batchredirect = self.get_value("batch_redirect", subgroup=None)
         submitcmd = ''
-        for string in (batchsubmit, submitargs, batchredirect, get_batch_script_for_job(job)):
-            if  string is not None:
-                submitcmd += string + " "
+        batch_env_flag = self.get_value("batch_env", subgroup=None)
```

Comment:
Important: if the current machine passes arguments to the batch script via batch env flag, then this needs to appear in the command line before the batch script name, otherwise it needs to come after.